### PR TITLE
Add Message Headers to Fluentd Event from Source #58

### PR DESCRIPTION
**Pull request by MaheshKReddy**
- Add a new source flag "include_headers" with default value false. When set to true, add Message headers to the payload with key "headers"

### CODE REVIEW

**Comment by MaheshKReddy** [comment]
@warmfusion :: Can you please review the Pullrequest ?

File: lib/fluent/plugin/in_amqp.rb [hunk 1]
```diff
@@ -100,20 +103,25 @@ def multi_workers_ready?
...
           end
         end
-        parsed
       else
-        { "message" => msg }
+        parsed = { 'message' => msg }
+      end
+      if @include_headers
+        log.debug 'Adding headers'
+        { 'headers' => meta[:headers] }.merge(parsed)
```
Comment (warmfusion) [inline lib/fluent/plugin/in_amqp.rb]: What happens if `headers` already exists in the incoming message body and isn't a hash? Do we perhaps need to allow the target field to be configurable, or simply explicitally state that any existing field value under `headers` will be merged (if hash) or replaced entirely (if anything else)?

Comment (warmfusion) [reply lib/fluent/plugin/in_amqp.rb]: @maheshkreddy - Can you let me know what you think would be best?

I'd probably go with simplicity and stick with the merge but update the docs along with it?

Comment (MaheshKReddy) [reply lib/fluent/plugin/in_amqp.rb]: @warmfusion I agree with your suggestion of allowing to have configurable field to set header field name. But, as i am on crunch with existing projects, I will go ahead and update the docs for now.

Comment (MaheshKReddy) [reply lib/fluent/plugin/in_amqp.rb]: @warmfusion Updated the docs

**Pull request merged**